### PR TITLE
added poet challenge persistentcy

### PR DESCRIPTION
**Pull request by antonlerner**

### CODE REVIEW

File: activation/activation.go [hunk 1]
```diff
@@ -159,80 +170,136 @@ func (b *Builder) loop() {
...
-		atxPubLayerID := types.LayerID(0)
-		if err == nil {
-			posAtxEndTick = posAtx.EndTick
-			b.posLayerID = posAtx.PubLayerIdx
-			atxPubLayerID = b.posLayerID.Add(b.layersPerEpoch)
-			posAtxId = posAtx.Id()
-		} else {
-			if !epoch.IsGenesis() {
-				return false, fmt.Errorf("cannot find pos atx: %v", err)
+		if b.challenge == nil {
```
Comment (noamnelke) [inline activation/activation.go]: I would put this before `if b.challenge != nil {` and then use `else`

**Review by noamnelke — APPROVED** [review]

File: activation/activation.go [hunk 2]
```diff
@@ -159,80 +170,136 @@ func (b *Builder) loop() {
...
+func (b *Builder) discardChallenge() error {
+	return b.store.Put(b.getNipstKey(), []byte{})
+}
+
 // PublishActivationTx attempts to publish an atx for the given epoch, it returns an error if an atx cannot be created
 // publish atx may not produce an atx each time it is called, that is expected behaviour as well.
-func (b *Builder) PublishActivationTx(epoch types.EpochId) (bool, error) {
+func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
 	if b.nipst != nil {
 		b.log.Info("re-entering atx creation in epoch %v", epoch)
```
Comment (barakshani) [inline activation/activation.go]: If it's not too much of a hassle, please change this log (and the one below) to something like "... in epoch j, for epoch k"

Comment (noamnelke) [reply activation/activation.go]: if you're changing this, extract the values to fields and make the log message constant.

Comment (antonlerner) [reply activation/activation.go]: There is a print that says this exactly if we have a challenge

**Pull request merged**